### PR TITLE
feat(githubconnector, aircallconnector): added ratelimit info extraction & handling

### PR DESCRIPTION
**Pull request by raphaelvignes**
## Change Summary

This PR aims to add ratelimit handling feature to Github & Aircall Connectors. 
The goal is to be able to know when the extraction reached the API Rate Limit and pause until the limit is reset. 

For Github see: https://docs.github.com/en/graphql/overview/resource-limitations
For Aircall See: https://developer.aircall.io/api-references/#rate-limiting

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%

### CODE REVIEW

File: toucan_connectors/aircall/aircall_connector.py [hunk 1]
```diff
@@ -58,80 +64,102 @@ async def fetch_page(
...
+                    limit,
+                    current_pass,
+                    next_page,
+                    query_params=query_params,
+                )
+
+    except AircallRateLimitExhaustedException as a:
+        reset_timestamp = int(a.args[0])
+        delay = reset_timestamp - (int(datetime.timestamp(datetime.utcnow())) + 1)
+        logging.getLogger(__file__).info(f'Rate limit reached, pausing {delay} seconds')
```
Comment (fspot) [inline toucan_connectors/aircall/aircall_connector.py:138]: En général on met `__name__` en nom de logger (`__name__` est un nom de module tandis que `__file__` est le chemin du fichier)

File: tests/aircall/test_async_funcs.py [hunk 2]
```diff
@@ -168,3 +195,18 @@ async def test_query_params_not_named_arg(mocker):
...
+    an AircallRateLimitExhaustedException exception is throw by fetch
+    """
+    ds = AircallDataset('calls')
+    fake_data = {'data': {'stuff': 'stuff'}}
+    mocker.patch(
+        fetch_fn_name, side_effect=[AircallRateLimitExhaustedException('16100000'), fake_data]
+    )
+    mockedsleep = mocker.patch('toucan_connectors.aircall.aircall_connector.time.sleep')
+    await fetch_page(ds, [], 'session', 0, 0, 0, 0)
+    mockedsleep.assert_called_once()
```
Comment (fspot) [inline tests/aircall/test_async_funcs.py:212]: Possible de tester la valeur du sleep (`assert_called_once_with(xxx)`) histoire de checker que le calcul du temps d'attente est bon ?

File: toucan_connectors/aircall/aircall_connector.py (hunk 1 above)
Comment (raphaelvignes) [reply toucan_connectors/aircall/aircall_connector.py:138]: Je me trompe toujours, merci !

File: tests/aircall/test_async_funcs.py (hunk 2 above)
Comment (raphaelvignes) [reply tests/aircall/test_async_funcs.py:212]: je rajoute ça oui

Comment (raphaelvignes) [reply tests/aircall/test_async_funcs.py:219]: fait

**Review by fspot — APPROVED** [review]

**Comment by raphaelvignes** [comment]
Merci !

**Pull request merged**